python/pyarrow/tests/test_array.py

```diff
@@ -108,6 +108,22 @@ def test_long_array_format():
     assert result == expected
 
 
+def test_indented_string_format():
+    arr = pa.array(['', None, 'foo'])
+    result = arr.to_string(indent=1)
+    expected = '[\n "",\n null,\n "foo"\n]'
```

Comment:
I suggest we use multinline strings for the expectation, it's a bit easer to read for people getting to this test